src/subclass/progressbar.h

```diff
@@ -0,0 +1,114 @@
+#ifndef PROGRESSBAR_H
+#define PROGRESSBAR_H
+
+
+#include <QGraphicsDropShadowEffect>
+#include <QLabel>
+#include <QMouseEvent>
+#include <QProgressBar>
+#include <QPropertyAnimation>
+#include <QPushButton>
+#include <QVBoxLayout>
+#include <QWidget>
+
+
+class ProgressPainter;
+class ProgressBar : public QProgressBar
+{
+    Q_OBJECT
+public:
+    enum class Mode {
+        Definite,
+        Indefinite
+    };
+    ProgressBar(QWidget *parent = Q_NULLPTR);
+    ~ProgressBar();
+    QPushButton *confirmButton();
+    QPushButton *cancelButton();
+    QPushButton *confirm;
+    QPushButton *cancel;
+
+
+    void clearButtonConnection();
+private:
+    int _width = 360 * logicalDpiX() / 100;
+    Mode mode;
+    QLabel *title;
+    QWidget *content;
+    QWidget *widget;
+    ProgressPainter *proPainter;
+    QVBoxLayout *contentLayout;
+    QVBoxLayout *layout;
+    QPoint oldPos;
+    QPushButton *closeBtn;
+
+
+    QString confirmationText;
+    QGraphicsOpacityEffect *opacity;
+
+
+    bool animating;
+    bool showingCancelDialog;
+    bool canClose;
+    void configureUI();
+    void configureConnection();
+    void updateMode();
+signals:
+    void confirmButtonClicked();
+    void cancelButtonClicked();
+public slots:
+    void setBackgroundColor(QColor color);
+    void showConfirmationDialog();
+    void hideConfirmationDialog();
+    void setConfirmationText(QString string);
+    void setConfirmationButtons(QString confirm, QString cancel, bool showConfirm = true, bool showCancel = true);
+    void showConfirmButton(bool showConfirm);
+    void showCancelButton(bool showCancel);
+    void setMode(ProgressBar::Mode mode);
+    void setTitle(QString string);
+    void close();
```

Comment:
also reimplement `exec()` which is what we will actually use to create a blocking widget when loading